src/DIRAC/ResourceStatusSystem/Agent/RucioRSSAgent.py

```diff
@@ -26,38 +32,122 @@ class RucioRSSAgent(AgentModule):
   """
   .. class:: RucioRSSAgent
 
-  Agent that synchronizes Rucio and Dirac
+  Agent that passes Dirac SE status to Rucio.
   """
 
-  def execute(self):
-    """ execution in one agent's cycle
+  def initialize(self):
+    """ agent's initialisation
 
     :param self: self reference
     """
+    self.log = gLogger.getSubLogger('RucioRSS')
+    self.log.info("Starting RucioRSSAgent")
+    # CA location
+    self.caCertPath = Locations.getCAsLocation()
+    # configured VOs
+    res = getVOs()
+    if res['OK']:
+      voList = getVOs().get('Value', [])
+    else:
+      return S_ERROR(res['Message'])
+    # configHelper test
+    if isinstance(voList, str):
+      voList = [voList]
+
+    self.clientConfig = configHelper(voList)
+    self.log.debug(" VO-specific Rucio Client config parameters: ", self.clientConfig)
+    return S_OK()
+
+  def execute(self):
+    """
+    Perform RSS->RSE synchronisation for all eligible VOs.
+
+    :return: S_OK or S_ERROR
+    """
+    voRes = {}
+    for key in self.clientConfig:
+      result = self.executeForVO(key)
+      voRes[key] = result['OK']
```

Comment:
Shouldn't the last line be `voRes[key] = result['Value']`?

And also you can use dict comprehension here.

Reply:
No, it should be `result['OK']` . It is meant to record failed attempts for a VO. It was not very smart, though. It would be enough only to record failed VOs. What about this ?

```python 
 gen = ((key, self.executeForVO(key)) for key in self.clientConfig)
 voRes = {key: result['Message'] for key, result in self.gen if not result['OK']}
```

This way we record the message as well. Now below, we do:

```python
   if nor voRes:
      return S_OK()
    else:
      # print message with a dict of _vo:Message_
```

And it uses a dict comprehension ;-)

Reply:
Seems better.